spec/lib/generators/action_generator_advanced_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'spec_helper'
+
+require_relative '../../../lib/generators/light_service/action_generator.rb'
+require_relative './full_generator_test_blobs'
+
+describe LightService::Generators::ActionGenerator, :type => :generator do
+  destination File.expand_path('tmp', __dir__)
+
+  context "when generating an advanced action" do
+    before(:all) do
+      prepare_destination
+      run_generator
+    end
+
+    after(:all) do
+      FileUtils.rm_rf destination_root
+    end
+
+    arguments %w[
+      my/fancy/action
+      expects:foo,bar
+      promises:baz,qux
+      --no-roll-back
+      --dir=services
+    ]
+
+    specify do
+      expect(destination_root).to(have_structure do
+        directory "app/services/my/fancy" do
```

Comment:
I looked at the directory structure after I ran the tests and I noticed that these tests are "polluting" the code's directory structure. 

Take a look at this:

```
% tree app
app
├── actions
├── organizers
├── processes
│   └── my
│       └── fancy
└── services
    └── my
        └── fancy

8 directories, 0 files
```

And this:

```
% tree spec/services
spec/services
└── my
    └── fancy

2 directories, 0 files
```

I am not happy with the tests creating these directories. Can you use the host OS' `/tmp` dir for this purpose?

Reply:
Hmmm... good point. I can totally try, it might be an option in the `generator_spec` library I was using. I'll dig into its code and see if this is possible.

I do clean up after each test run though with an `after` block.